File: src/test/java/eotree/TestEOMetas.java
package eotree;

import arrow.core.Some;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static util.ListUtils.listOf;

public class TestEOMetas {

    @Test
    public void TestGenerateEOZeroIndent() {
        var f = new EOMetas(
                new Some<>("org.eolang.package"),
                listOf(
                        new EOMeta("key1", "value1"),
                        new EOMeta("key2", "value2")
                )
        );
        assertEquals("""
                        +package org.eolang.package
                        +key1 value1
                        +key2 value2""",
                f.generateEO(0));
    }

    @Test
    public void TestGenerateEONonZeroIndent() {
        var f = new EOMetas(
                new Some<>("org.eolang.package"),
                listOf(
                        new EOMeta("key1", "value1"),
                        new EOMeta("key2", "value2")
                )
        );
        assertEquals("""
                        +package org.eolang.package
                        +key1 value1
                        +key2 value2""",
                f.generateEO(1));
    }
}

File: src/test/kotlin/eotree/TestEOMetas.kt
package eotree

import arrow.core.Some
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.polystat.j2eo.eotree.EOMeta
import org.polystat.j2eo.eotree.EOMetas
import org.polystat.j2eo.util.ListUtils.listOf

/**
 * EO metas tests.
 */
class TestEOMetas {
    @Test
    fun testGenerateEOZeroIndent() {
        val f = EOMetas(
            Some("org.eolang.package"),
            listOf(
                EOMeta("key1", "value1"),
                EOMeta("key2", "value2")
            )
        )
        Assertions.assertEquals(
            "+package org.eolang.package\n" +
                "+key1 value1\n" +
                "+key2 value2",
            f.generateEO(0)
        )
    }

    @Test
    fun testGenerateEONonZeroIndent() {
        val f = EOMetas(
            Some("org.eolang.package"),
            listOf(
                EOMeta("key1", "value1"),
                EOMeta("key2", "value2")
            )
        )
        Assertions.assertEquals(
            "+package org.eolang.package\n" +
                "+key1 value1\n" +
                "+key2 value2",
            f.generateEO(1)
        )
    }
}
